examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,133 +60,161 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
 }
 """
 function Initialise_DYCOMS!(state::Vars, aux::Vars, (x,y,z), t)
-  DT         = eltype(state)
-  xvert::DT  = z
-  
-  epsdv::DT     = molmass_ratio
-  q_tot_sfc::DT = 8.1e-3
-  Rm_sfc::DT    = gas_constant_air(PhasePartition(q_tot_sfc))
-  ρ_sfc::DT     = 1.22
-  P_sfc::DT     = 1.0178e5
-  T_BL::DT      = 285.0
-  T_sfc::DT     = P_sfc/(ρ_sfc * Rm_sfc);
-  
-  q_liq::DT      = 0
-  q_ice::DT      = 0
-  zb::DT         = 600   
-  zi::DT         = 840 
+  FT            = eltype(state)
+  xvert::FT     = z
+  #These constants are those used by Stevens et al. (2005)
+  qref::FT      = 7.75e-3
+  q_tot_sfc::FT = qref
+  q_pt_sfc      = PhasePartition(q_tot_sfc)
+  Rm_sfc        = gas_constant_air(q_pt_sfc)
+  T_sfc::FT     = 292.5
+  P_sfc::FT     = MSLP
+  ρ_sfc::FT     = P_sfc / Rm_sfc / T_sfc
+  # Specify moisture profiles
+  q_liq::FT      = 0
+  q_ice::FT      = 0
+  zb::FT         = 600    # initial cloud bottom
+  zi::FT         = 840    # initial cloud top
   dz_cloud       = zi - zb
-  q_liq_peak::DT = 4.5e-4
-  
-  if xvert > zb && xvert <= zi        
+  q_liq_peak::FT = 0.00045 #cloud mixing ratio at z_i
+  if xvert > zb && xvert <= zi
     q_liq = (xvert - zb)*q_liq_peak/dz_cloud
   end
-  if ( xvert <= zi)
-    θ_liq  = DT(289)
-    q_tot  = DT(8.1e-3)
+  if xvert <= zi
+    θ_liq = FT(289)
+    q_tot = qref
   else
-    θ_liq = DT(297.5) + (xvert - zi)^(DT(1/3))
-    q_tot = DT(1.5e-3)
+    θ_liq = FT(297.5) + (xvert - zi)^(FT(1/3))
+    q_tot = FT(1.5e-3)
   end
+  # --------------------------------------------------
+  # perturb initial state to break the symmetry and
+  # trigger turbulent convection
+  # --------------------------------------------------
+  randnum1   = rand(seed, FT) / 100
+  randnum2   = rand(seed, FT) / 1000
+  #randnum1   = rand(Uniform(-0.02,0.02), 1, 1)
+  #randnum2   = rand(Uniform(-0.000015,0.000015), 1, 1)
+  if xvert <= 25.0
+    θ_liq += randnum1 * θ_liq
+    #q_tot += randnum2 * q_tot
+  end
+  # --------------------------------------------------
+  # END perturb initial state
+  # --------------------------------------------------
 
-  q_pt = PhasePartition(q_tot, q_liq, DT(0))
-  Rm    = gas_constant_air(q_pt)
-  cpm   = cp_m(q_pt)
+  # Calculate PhasePartition object for vertical domain extent
+  q_pt  = PhasePartition(q_tot, q_liq, q_ice)
   #Pressure
-  H = Rm_sfc * T_BL / grav;
-  P = P_sfc * exp(-xvert/H);
-  #Exner
-  exner_dry = exner(P, PhasePartition(DT(0)))
-  #Temperature 
-  T             = exner_dry*θ_liq + LH_v0*q_liq/(cpm*exner_dry);
-  #Density
-  ρ             = P/(Rm*T);
-  #Potential Temperature
-  θv     = virtual_pottemp(T, P, q_pt)
-  # energy definitions
-  u, v, w     = DT(7), DT(-5.5), DT(0)
-  U           = ρ * u
-  V           = ρ * v
-  W           = ρ * w
-  e_kin       = DT(1//2) * (u^2 + v^2 + w^2)
+  H     = Rm_sfc * T_sfc / grav;
+  p     = P_sfc * exp(-xvert/H);
+  #Density, Temperature
+  TS    = LiquidIcePotTempSHumEquil_no_ρ(θ_liq, q_pt, p)
```

Comment:
This will need to be updated after PR #506 is merged. @charleskawczynski can help.

Reply:
> If you can get `T` from `theta_l` and `q_t` using your functions then you can get `rho` from the equation of state using `p` and `T` as I do below in my matlab script (@tapios please, confirm that the operations below would be correct here):
> 
> ```
>   `p = p_sfc * exp(-z / H)
> 
> %Exner
> exner = (p / p_sfc)^(R_d/cp_d);
> 

This looks ok.

> %T, Tv 
> T     = exner*theta_liq + Lv0 * q_liq / (cpm*exner);

The second term should not be divided by the Exner function: Just 
> T     = exner*theta_liq + Lv0 * q_liq / (cpm);

(I put the derivations in the design docs; please refer to those.) 

@charleskawczynski I think providing a function `LiquidIcePotTempSHumEquil_given_pressure(θ_liq, q_pt, p)` would be useful. In essence, this is how @smarras79 is calculating `T`.

> %Density
> rho  = p/(Rm*T);
> `
> ```

Reply:
Just for reference, we'll have to use `LiquidIcePotTempSHumNonEquil_given_pressure` for now until we come up with a consistent constructor for an equilibrium phase.

Reply:
This driver is now outdated and needs to be replaced with the currently running one.